encoding/codecs/h264.md

```diff
@@ -0,0 +1,247 @@
+
+Beyond the preset chosen,
+x264 allows us to further tune the encoding settings
+for the type of content we're working with.
+The following tunings are generally the most useful:
+
+- `film`: Recommended for live action videos.
+- `animation`: Recommended for anime or cartoons with flat textures.
+For 3D animation (e.g. modern Pixar movies),
+you may find better results with `film`.
+- `grain`: Recommended for particularly grainy films.
+
+You don't need to use a tuning,
+but it generally helps to produce a better-looking video.
+
+#### `--crf 24`
+
+CRF is a constant-quality, 1-pass encoding mode.
+In layman's terms,
+this means that we don't need the output to meet a specific filesize,
+we just want the output to meet a certain quality level.
+CRF ranges from 0 to 51,
+with 0 being the best quality and 51 being the smallest filesize,
+but there is a certain range of CRF settings
+that are generally most useful.
+Here are some guidelines:
+
+- CRF 13: This is considered visually lossless to videophiles.
+This can produce rather large files,
+but is a good choice if you want high quality videos.
+Fansubbing groups often use this for Blu-ray encodes.
+- CRF 16-18: This is considered visually lossless to most viewers,
+and leans toward high quality while still providing a reasonable filesize.
+This is a typical range for fansub encodes.
+- CRF 21-24: This provides a good balance between quality and filesize.
+Some quality loss becomes visible, but this is generally a good choice
+where filesize becomes a concern, such as for videos viewed over the internet.
+- CRF 26-30: This prioritizes filesize, and quality loss becomes more obvious.
+It is generally not recommended to go higher than CRF 30 in any real-world
+encoding scenario, unless you want your videos to look like they were made
+for dial-up.
+
+#### `-o x264output.mkv -`
+
+This last portion tells which files to use for the input and output.
+We use `-o` to tell which filename to write the encoded file to.
+In this case, x264 will write a file at `x264output.mkv`
+in the current directory.
+
+The last argument to x264 is always the input file.
+In this case, we pass `-` for the input file,
+which tells x264 to use the piped output from vspipe.
+
+## Example 2
+
+For the next example, let's say we want to make sure our encode
+fits onto a single 4.7GB DVD. How would we do that in x264?
+
+First, we'll need to figure out what bitrate our encode should be,
+in **kilobits per second**. This means we'll need to know a couple of things:
+
+- The length of our video, in seconds.
+For this example, let's say our movie is 2 hours (120 minutes) long.
+We'll convert that to seconds: 120 \* 60 = **7200 seconds**.
+- Our target filesize.
+We know that this is 4.7GB,
+but we need to convert it to kilobits.
+We can do this with the following steps:
+  - 4.7GB \* 1024 = 4812.8MB
+  - 4812.8MB \* 1024 = 4928307.2KB (uppercase B = bytes)
+  - 4928307.2KB \* 8 = **39426457.6Kb** (lowercase b = bits)
```

Comment:
These equasions don't match the units on both sides. While some people think it's simpler to understand without the seemingly superfluous units, I prefer the explicit and correct sytle of `1024MB/GB` here, because it's self-documenting.

Also, while we're at it, these should be `GiB` and `MiB`.

Reply:
And I much prefer `Kbit` over `Kb`. That would probably make the explanation of uppercase/lowercase unnecessary as well.